compiler/src/dmd/main.d

```diff
@@ -488,29 +488,8 @@ private int tryMain(size_t argc, const(char)** argv, ref Param params)
     if (global.errors)
         fatal();
 
-    if (params.dihdr.doOutput)
-    {
-        /* Generate 'header' import files.
-         * Since 'header' import files must be independent of command
-         * line switches and what else is imported, they are generated
-         * before any semantic analysis.
-         */
-        OutBuffer buf;
-        foreach (m; modules)
-        {
-            if (m.filetype == FileType.dhdr)
-                continue;
-            if (params.v.verbose)
-                message("import    %s", m.toChars());
-
-            buf.reset();         // reuse the buffer
-            genhdrfile(m, params.dihdr.fullOutput, buf);
-            if (!writeFile(m.loc, m.hdrfile.toString(), buf[]))
-                fatal();
-        }
-    }
     if (global.errors)
-        removeHdrFilesAndFail(params, modules);
+        fatal();
```

Comment:
Why are we no longer removing header files here? Is it because they are not generated prior to this step?

Reply:
Also Walter has been removing a bunch of calls to fatal, but this is in `tryMain` so I'm not sure if that logic applies here.